### DISCUSSION TITLE
How to get a project to show up in dependency graph?

### DISCUSSION THREAD

**Discussion by jvdb** [community/community#22723] [New to GitHub]
<p>The Dependency graph for one of my Java projects looks like this: <a href="https://github.com/parsingdata/metal/network/dependencies" rel="nofollow noopener">https://github.com/parsingdata/metal/network/dependencies</a>, if you scroll all the way to the bottom you can see that the “formats”-module depends on the “metal-core”-module, but unfortunately that dependency is not a link to the repo? How can I fix this?</p>
<p>As far as I can see the Maven files look good, and Github did succeed in extracting the dependency, just not the link (which is the URL in the pom.xml…)</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Thanks for reaching out!</p>
<p>I took a look at your <a href="https://github.com/parsingdata/metal/blob/master/formats/pom.xml" rel="nofollow noopener">pom.xml</a> for the module you’re talking about. I’m not very experienced with Maven but I suspect I have an idea what the problem is.</p>
<p>If you look at the two dependencies listed in that pom.xml file, one is linked and the other isn’t. The one that is linked is a top-level GitHub repository <a href="https://github.com/hcoles/pi-test" rel="nofollow noopener">https://github.com/hcoles/pi-test</a>. The one that isn’t linked is not a top-level GitHub repository. There is no <a href="https://github.com/parsingdata/metal-core" rel="nofollow noopener">https://github.com/parsingdata/metal-core</a>, only <a href="https://github.com/parsingdata/metal.%C2%A0Since" rel="nofollow noopener">https://github.com/parsingdata/metal. Since</a> the feature is designed to link back to the GitHub repository associated with a dependency, it isn’t linking it because, technically, there isn’t one for that dependency.</p>
<p>As the feature is currently implemented, I don’t believe there is a way to fix it without extracting metal-core out into its own repository.</p>
<p>I hope that helps!</p>

**Comment by jvdb** [comment]
<p>Thanks for the response. Seems like a fair explanation for what we see indeed!</p>
<p>This makes me wonder whether I could fix it by giving the submodules their own URL-property in their pom.xml with a deeplink to their directory in the repository, so for metal-core it would point to <a href="https://github.com/parsingdata/metal/tree/master/core" rel="nofollow noopener">https://github.com/parsingdata/metal/tree/master/core</a>. But I will try this myself with a next release and see what happens.</p>

**Comment by jojomatik** [comment]
<p>Hi <a class="mention" href="/jvdb">@jvdb</a>,</p>
<p>I’m currently wondering how to set up a link from one of my repositories to another. As you said in your first post the dependency is picked up, but it’s not clickable/ linked.</p>
<p>Have you got your configuration working by now? And if so, what did you change?</p>
<p>Thank You.</p>
